app/uk/gov/hmrc/gform/formtemplate/JsonSchemeValidator.scala

```diff
@@ -67,59 +43,193 @@ object JsonSchemeValidator {
   def validateJson(json: Json): Either[SchemaValidationException, Unit] =
     parsedSchema match {
       case Left(parsingFailure) => Left(SchemaValidationException("Schema error: " + parsingFailure))
-      case Right(schema) =>
+      case Right(schema: Json) =>
```

Comment:
Do not repeat types in pattern matching. This is for consistency, since we don't do this in other places